### PR TITLE
Inferred service dependencies guide changes

### PR DESCRIPTION
**Pull request by dussault-antoine**
### What does this PR do? What is the motivation?
This PR clarifies what the beta is about and what are the breaking changes entailed by opting-in to the experience.

### Merge instructions

- [ ] Please merge after reviewing

### CODE REVIEW

File: content/en/service_catalog/guides/troubleshooting.md [hunk 1]
```diff
@@ -16,10 +16,6 @@ If services that you know are instrumented for APM are not appearing in the Serv
 
 {{< img src="tracing/service_catalog/svc_cat_troubleshooting_1.png" alt="Hover message indicating that no performance data has been reported in the past hour" >}}
 
-## Extra services are discovered as an artifact of APM instrumentation
```
Comment (pkalmakis) [inline content/en/service_catalog/guides/troubleshooting.md:19]: It's still true that you can remove these along with opting into the beta. We just aren't making this a requirement of the beta. I don't think we need to remove this.

File: content/en/tracing/guide/inferred-service-opt-in.md [hunk 2]
```diff
@@ -16,30 +16,34 @@ further_reading:
...
+{{< img src="tracing/services/service_page/dependencies.png" alt="Service page service dependency map" style="width:100%;">}}
 
 ## Opt in
 
-To opt in, you must adjust your Datadog Agent and APM Tracer configurations. Check the [Global default service naming migration](#global-default-service-naming-migration), to see if you need to take any migration actions.
+<div class="alert alert-warning">Only go through migration steps once Datadog support confirmed the feature is enabled for you on the Datadog side.</div>
+
+To opt in, you must adjust:
+- your [Datadog Agent](#datadog-agent-configuration) (or [OpenTelemetry collector](#opentelemetry-collector)) configuration
+- your [APM tracing libraries](#apm-tracing-libary-configuration) configuration
```
Comment (pkalmakis) [inline content/en/tracing/guide/inferred-service-opt-in.md:46]: This is optional but recommended. If a customer doesn't want to take on the breaking changes, the experience still works but not as well.

File: content/en/tracing/guide/inferred-service-opt-in.md [hunk 3]
```diff
@@ -16,30 +16,34 @@ further_reading:
...
 {{< callout url="https://docs.google.com/forms/d/1imGm-4SfOPjwAr6fwgMgQe88mp4Y-n_zV0K3DcNW4UA/edit" d_target="#signupModal" btn_hidden="true" btn_hidden="false" header="Opt in to the private beta!" >}}
-Inferred entities is in private beta. To request access, complete the form.
+Inferred service dependencies are in private beta. To request access, complete the form.
 {{< /callout >}}
 
-Follow the steps below to enable the new dependency map on the [Service Page][1], and add _inferred services_ to the [Service Map][2] and [Service Catalog][3] pages.
+## Inferred service dependencies
 
-## Inferred entities
+Datadog can automatically discover the dependencies for an instrumented service, such as a database, a queue, or a third-party API even if that dependency hasn't been instrumented yet. Datadog infers the presence of these dependencies and collects its associated performance metrics based on information on the outbound requests from your instrumented services.
```
Comment (brett0000FF) [inline content/en/tracing/guide/inferred-service-opt-in.md:24]:
```suggestion
Datadog can automatically discover the dependencies for an instrumented service, such as a database, a queue, or a third-party API, even if that dependency hasn't been instrumented yet. By analyzing outbound requests from your instrumented services, Datadog infers the presence of these dependencies and collects associated performance metrics.
```
Minor suggestions for clarity.

File: content/en/tracing/guide/inferred-service-opt-in.md [hunk 4]
```diff
@@ -16,30 +16,34 @@ further_reading:
...
 
-Datadog can automatically discover the dependencies for an instrumented service, such as a database, a queue, or a third-party API even if that dependency hasn't been instrumented yet. Datadog infers the presence of these dependencies and their performance based on information on the outbound requests of your instrumented services.
+With the new inferred entities experience, filter [Service Catalog][3] entries by entity type (database, queue, third-party API, etc), and better visualize service dependencies from the new [Service Page dependency map](#service-page-dependency-map) and other places in the APM product experience.
```
Comment (brett0000FF) [inline content/en/tracing/guide/inferred-service-opt-in.md:26]:
```suggestion
With the new inferred entities experience, you can filter [Service Catalog][3] entries by entity type, such as database, queue, or third-party API. This allows you to better visualize service dependencies using the [Service Page dependency map](https://github.com/DataDog/documentation/pull/23219/files#service-page-dependency-map) and APM features.
```

File: content/en/tracing/guide/inferred-service-opt-in.md [hunk 5]
```diff
@@ -16,30 +16,34 @@ further_reading:
 ---
 
 {{< callout url="https://docs.google.com/forms/d/1imGm-4SfOPjwAr6fwgMgQe88mp4Y-n_zV0K3DcNW4UA/edit" d_target="#signupModal" btn_hidden="true" btn_hidden="false" header="Opt in to the private beta!" >}}
-Inferred entities is in private beta. To request access, complete the form.
+Inferred service dependencies are in private beta. To request access, complete the form.
 {{< /callout >}}
 
-Follow the steps below to enable the new dependency map on the [Service Page][1], and add _inferred services_ to the [Service Map][2] and [Service Catalog][3] pages.
+## Inferred service dependencies
```
Comment (brett0000FF) [inline content/en/tracing/guide/inferred-service-opt-in.md:22]:
```suggestion
## Overview
```

File: content/en/tracing/guide/inferred-service-opt-in.md (hunk 2 above)
Comment (brett0000FF) [inline content/en/tracing/guide/inferred-service-opt-in.md:46]:
```suggestion
To opt in, Datadog recommends you adjust your:
- [Datadog Agent](#datadog-agent-configuration) (or [OpenTelemetry collector](#opentelemetry-collector)) configuration
- [APM tracing libraries](#apm-tracing-libary-configuration) configuration
```
Simplify repetition in bullets and indicate this isn't required per Peter's note.

File: content/en/tracing/guide/inferred-service-opt-in.md [hunk 6]
```diff
@@ -92,7 +96,9 @@ exporters:
 {{< /code-block >}}
 
 
-### APM tracer configuration
+### APM tracing libary configuration
+
+<div class="alert alert-warning">Going through the following step is a <b>breaking change</b>: Datadog will change the way service names are captured by default. Check the <a href="https://docs.datadoghq.com/tracing/guide/inferred-service-opt-in#global-default-service-naming-migration">Global default service naming migration</a>, to see if you need to take any migration actions.</div>
```
Comment (brett0000FF) [inline content/en/tracing/guide/inferred-service-opt-in.md:101]:
```suggestion
<div class="alert alert-warning">The following steps introduce a <b>breaking change</b>: Datadog will change the way service names are captured by default. Refer to <a href="#global-default-service-naming-migration">Global default service naming migration</a>, to determine if you need to take any migration actions.</div>
```

File: content/en/tracing/guide/inferred-service-opt-in.md [hunk 7]
```diff
@@ -198,29 +204,46 @@ To opt in, add the following environment variables to your tracer settings or sy
 
 {{< /tabs >}}
 
+
+
+## The new nomemclature: what is changing
```
Comment (brett0000FF) [inline content/en/tracing/guide/inferred-service-opt-in.md:209]:
```suggestion
## The new nomenclature: What is changing
```

File: content/en/tracing/guide/inferred-service-opt-in.md [hunk 8]
```diff
@@ -198,29 +204,46 @@ To opt in, add the following environment variables to your tracer settings or sy
...
 ### Global default service naming migration
 
 When you enable the `DD_TRACE_REMOVE_INTEGRATION_SERVICE_NAMES_ENABLED` environment variable, it improves how service-to-service connections and inferred services are represented in Datadog visualizations, across all supported tracing library languages and integrations.
 
 Previously, some tracing libraries included the name of the associated integration in service name tagging. For example, .NET tagged gRPC calls as `service:<DD_SERVICE>-grpc-client` while Python tagged them as `service:grpc-client`. With this option enabled, all supported tracing libraries tag spans from the downstream services with the calling service's name, `service:<DD_SERVICE>`, thereby providing a _global default service name_.
 
+_ | Before | After
+--|-------|--------
+Service name | `service:my-service-grpc-client` or `service:grpc-client` | `service:myservice` 
+additional `peer.*` attributes | _No `peer.*` tags set_ | `@peer.service:otherservice` (`otherservice` being the name of the remote service being called via gRPC)
```
Comment (brett0000FF) [inline content/en/tracing/guide/inferred-service-opt-in.md:223]:
```suggestion
additional `peer.*` attributes | _No `peer.*` tags set_ | `@peer.service:otherservice` (`otherservice` being the name of the remote service being called with gRPC)
```

File: content/en/tracing/guide/inferred-service-opt-in.md [hunk 9]
```diff
@@ -198,29 +204,46 @@ To opt in, add the following environment variables to your tracer settings or sy
...
 - APM metrics
 - APM custom span metrics
 - Trace analytics
 - Retention filters
 - Sensitive data scans
-- Monitors, dashboards, or notebooks that query those things
+- Monitors, dashboards, or notebooks 
 
-Update those items to use the global default service tag (`service:<DD_SERVICE>`) instead.
+... that target similar service names, update those items to use the global default service tag (`service:<DD_SERVICE>`) instead.
```
Comment (brett0000FF) [inline content/en/tracing/guide/inferred-service-opt-in.md:241]:
```suggestion
And these target similar service names, update those items to use the global default service tag (`service:<DD_SERVICE>`) instead.
```

**Review by brett0000FF — APPROVED** [review]
Looks good! Left a few suggestions. Please let me know if you have any questions.

File: content/en/tracing/guide/inferred-service-opt-in.md [hunk 10]
```diff
@@ -16,30 +16,34 @@ further_reading:
...
+With the new inferred entities experience, filter [Service Catalog][3] entries by entity type (database, queue, third-party API, etc), and better visualize service dependencies from the new [Service Page dependency map](#service-page-dependency-map) and other places in the APM product experience.
 
-To determine the names and types of the inferred entities, Datadog uses span attributes. Inferred external APIs use the default naming scheme `net.peer.name`. For example, `api.stripe.com`, `api.twilio.com`, `us6.api.mailchimp.com`. Inferred databases use the default naming scheme `db.instance`.
+To determine the names and types of the inferred service dependencies, Datadog uses standard span attributes and maps them to `peer.*` attributes - find the full list of `peer.*` attributes [here](#inferred-service-dependencies-nomemclature) -. Inferred external APIs use the default naming scheme `net.peer.name`. For example, `api.stripe.com`, `api.twilio.com`, `us6.api.mailchimp.com`. Inferred databases use the default naming scheme `db.instance`.
```
Comment (brett0000FF) [inline content/en/tracing/guide/inferred-service-opt-in.md:28]:
```suggestion
To determine the names and types of the inferred service dependencies, Datadog uses standard span attributes and maps them to `peer.*` attributes. For the full list of `peer.*` attributes, see [Inferred service dependencies nomenclature](#inferred-service-dependencies-nomemclature). Inferred external APIs use the default naming scheme `net.peer.name`. For example, `api.stripe.com`, `api.twilio.com`, `us6.api.mailchimp.com`. Inferred databases use the default naming scheme `db.instance`.
```

File: content/en/service_catalog/guides/troubleshooting.md (hunk 1 above)
Comment (dussault-antoine) [reply content/en/service_catalog/guides/troubleshooting.md:19]: Even though this is true, as we already have close to 200 candidates for the beta, I don't think we need to advertise the beta anymore in other places in the docs (yet) - at least until we are in a more stable state where we can enable the feature for larger batches of customers.

**Pull request merged**